### PR TITLE
Export import wallet

### PR DESCRIPTION
**Pull request by poonam-ghewande**
1. Added  Export / Import wallet  in  application

### CODE REVIEW

**Comment by sairanjitAW** [comment]
We need to change a few flows
- Create pin screen should be after selection of (create new wallet/Import wallet)

**Comment by poonam-ghewande** [comment]
Sure we will do the change in flow for (create new wallet/Import wallet) .

File: App.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import AgentProvider from '@aries-framework/react-hooks'
+import { Buffer } from 'buffer'
```
Comment (sairanjit) [inline App.tsx:2]: Can we remove this as we are already using **shim.js** ?

File: app/assets/img/backup-success1.svg
Comment (sairanjit) [inline app/assets/img/backup-success1.svg:1]: Delete this svg if not required

File: app/assets/img/success-backup1.png
Comment (sairanjit) [inline app/assets/img/success-backup1.png:1]: Delete this svg if not required

File: app/navigators/RootStack.tsx [hunk 2]
```diff
@@ -272,6 +316,10 @@ const RootStack: React.FC = () => {
           component={Developer}
           options={{ ...defaultStackOptions, title: t('Screens.Developer'), headerBackTestID: testIdWithKey('Back') }}
         />
+        <Stack.Screen name={Stacks.ContactStack} component={ContactStack} />
```
Comment (sairanjit) [inline app/navigators/RootStack.tsx:319]: Please add the Navigation stacks in the appropriate location,(Notification, Connection, ... etc)

File: app/navigators/AuthenticateStack.tsx [hunk 3]
```diff
@@ -15,12 +20,24 @@ interface AuthenticateStackProps {
...
   const Stack = createStackNavigator<AuthenticateStackParams>()
   const theme = useTheme()
+  const { t } = useTranslation()
   const defaultStackOptions = createDefaultStackOptions(theme)
 
   return (
     <Stack.Navigator screenOptions={{ ...defaultStackOptions, presentation: 'transparentModal', headerShown: false }}>
       <Stack.Screen name={Screens.EnterPIN} component={PINEnter} initialParams={{ setAuthenticated }} />
       <Stack.Screen name={Screens.AttemptLockout} component={AttemptLockout} />
+      <Stack.Screen
```
Comment (sairanjit) [inline app/navigators/AuthenticateStack.tsx:30]: Add (ImportWalletVerify, ImportSuccess, CreateWallet) in appropriate stack

File: app/screens/CreateWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+const CreateWallet: React.FC = () => {
+  const { TextTheme, ColorPallet } = useTheme()
+  const navigation = useNavigation()
+  const styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      height: '100%',
+    },
+    titletext: {
```
Comment (sairanjit) [inline app/screens/CreateWallet.tsx:17]:
```suggestion
    titleText: {
```

File: app/screens/CreateWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      fontSize: 32,
+      fontWeight: '500',
+      alignSelf: 'center',
+      color: ColorPallet.brand.primary,
+      marginTop: 20,
+    },
+    subcontainer: {},
```
Comment (sairanjit) [inline app/screens/CreateWallet.tsx:24]:
```suggestion
    }
```

File: app/screens/CreateWallet.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  })
+  return (
+    <View style={styles.container}>
+      <Text style={(TextTheme.headingTwo, styles.titletext)}>Welcome</Text>
```
Comment (sairanjit) [inline app/screens/CreateWallet.tsx:28]:
```suggestion
      <Text style={(TextTheme.headingTwo, styles.titleText)}>Welcome</Text>
```

File: app/screens/ExportWallet.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+const ExportWallet: React.FC = () => {
+  const { ColorPallet, TextTheme } = useTheme()
+  const navigation = useNavigation()
+  const { t } = useTranslation()
+  const [phraseData, setPhraseData] = useState([])
+  useEffect(() => {
+    const mnemonic = generateMnemonic(128)
+    const mnemonicArray = mnemonic.split(' ')
+
+    const mnemonicIndividualWordsArray: any = []
```
Comment (sairanjit) [inline app/screens/ExportWallet.tsx:21]:
```suggestion
    const mnemonicIndividualWordsArray: string[] = []
```

File: app/screens/ExportWalletConfirmation.tsx [hunk 8]
```diff
@@ -0,0 +1,261 @@
+import { useAgent } from '@aries-framework/react-hooks'
+import { useNavigation, useRoute } from '@react-navigation/core'
+import md5 from 'md5'
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { View, Text, TouchableOpacity, ScrollView, PermissionsAndroid, Platform, Share } from 'react-native'
+import RNFS, { exists, mkdir, unlink, DownloadDirectoryPath } from 'react-native-fs'
```
Comment (sairanjit) [inline app/screens/ExportWalletConfirmation.tsx:7]: remove unused imports

File: app/screens/ExportWalletConfirmation.tsx [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+    const updatedArraySetPhraseData = [...arraySetPhraseData]
+    updatedArraySetPhraseData[index] = item
+    setPhraseData(updatedPhraseData)
+    setArraySetPhraseData(updatedArraySetPhraseData)
+
+    setNextPhraseIndex(index)
+  }
+
+  const setPhrase = (item: string, index: number) => {
```
Comment (sairanjit) [inline app/screens/ExportWalletConfirmation.tsx:156]: unused parameter

File: app/screens/ImportSuccess.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  const { TextTheme } = useTheme()
+  const navigation = useNavigation()
+  const styles = StyleSheet.create({
+    container: {
+      justifyContent: 'center',
+      height: '100%',
+      flex: 1,
+      alignSelf: 'center',
+    },
+    labeltext: {
```
Comment (sairanjit) [inline app/screens/ImportSuccess.tsx:19]:
```suggestion
    labelText: {
```

File: app/screens/ImportSuccess.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+      CommonActions.reset({
+        index: 0,
+        routes: [{ name: Screens.Credentials }],
+      }),
+    )
+  })
+  return (
+    <View style={styles.container}>
+      <Assets.svg.BackupSuccess height={225} width={230} />
+      <Text style={(TextTheme.labelText, styles.labeltext)}>Wallet Restored successfully</Text>
```
Comment (sairanjit) [inline app/screens/ImportSuccess.tsx:37]:
```suggestion
      <Text style={(TextTheme.labelText, styles.labelText)}>Wallet Restored successfully</Text>
```

File: app/screens/ImportWalletConfirmation.tsx
Comment (sairanjit) [inline app/screens/ImportWalletConfirmation.tsx:1]: Remove unused imports

File: app/screens/Success.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+        index: 0,
+        routes: [{ name: Stacks.TabStack }],
+      }),
+    )
+  }, [])
+  return (
+    <View style={styles.container}>
+      <Assets.svg.BackupSuccess height={225} width={230} style={{ alignSelf: 'center', alignItems: 'center' }} />
+      <Text style={(TextTheme.labelText, styles.labeltext)}>Exported successfully</Text>
+      {/* <Text style={styles.namefiletext}>{parms?.params?.encryptedFileLocation}</Text> */}
```
Comment (sairanjit) [inline app/screens/Success.tsx:45]: remove commented code

File: app/screens/Success.tsx
Comment (sairanjit) [inline app/screens/Success.tsx:1]: remove unused variables and use camelCase for all the styles

File: app/utils/crystals-kyber.ts
Comment (sairanjit) [inline app/utils/crystals-kyber.ts:1]: Remove the crystal Kyber package

File: app/WalletExportstyle.ts
Comment (sairanjit) [inline app/WalletExportstyle.ts:1]: Keep styles in the same file of the screen

File: App.tsx
Comment (sairanjit) [inline App.tsx:1]: remove buffer as we are using rn-nodify

File: package.json
Comment (sairanjit) [inline package.json:1]: remove unused dependencies

**Review by sairanjit — CHANGES_REQUESTED** [review]

File: app/assets/img/backup-success1.svg
Comment (poonam-ghewande) [reply app/assets/img/backup-success1.svg:1]: Deleted  the svg .

File: app/assets/img/success-backup1.png
Comment (poonam-ghewande) [reply app/assets/img/success-backup1.png:1]: Deleted the unused svg

File: app/WalletExportstyle.ts
Comment (poonam-ghewande) [reply app/WalletExportstyle.ts:1]: Added styles in same screen.

File: app/navigators/AuthenticateStack.tsx (hunk 3 above)
Comment (poonam-ghewande) [reply app/navigators/AuthenticateStack.tsx]: Added   (ImportWalletVerify, ImportSuccess, CreateWallet)  in setting stack.

File: app/screens/Success.tsx
Comment (poonam-ghewande) [reply app/screens/Success.tsx:1]: removed.

File: app/utils/crystals-kyber.ts
Comment (poonam-ghewande) [reply app/utils/crystals-kyber.ts:1]: Removed crystal kyber package.

File: package.json
Comment (poonam-ghewande) [reply package.json:1]: Removed unused dependencies.

File: app/navigators/RootStack.tsx (hunk 2 above)
Comment (poonam-ghewande) [reply app/navigators/RootStack.tsx]: Updated the  Navigation stack in RootStack .

**Review by sairanjit — APPROVED** [review]

**Pull request merged**